### PR TITLE
Revert "Ensure physical host buttons only perform one bound event"

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

Reverts 

- https://github.com/dosbox-staging/dosbox-staging/pull/4079/commits

to fix

- https://github.com/dosbox-staging/dosbox-staging/issues/4128

Reopens

- https://github.com/dosbox-staging/dosbox-staging/issues/4051

## Related issues

- https://github.com/dosbox-staging/dosbox-staging/issues/4126

# Release notes

N/A

# Manual testing

The change has been manually tested on:

- [ ] Windows
- [ ] macOS
- [ ] Linux

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [ ] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [ ] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] my change has been manually tested on Windows, macOS, and Linux.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] provided the release notes draft (for significant user-facing changes).
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/gui/sdl_mapper.cpp [hunk 1]
```diff
@@ -266,9 +266,6 @@ class CBind {
 	}
 
 	void ActivateBind(Bits _value,bool ev_trigger,bool skip_action=false) {
-		if (!event) {
```
Comment (arifogel) [inline src/gui/sdl_mapper.cpp:269]: Why revert this part (from this [commit](https://github.com/dosbox-staging/dosbox-staging/pull/4129/commits/5653d2ad1586d43bfc4fa3e36d7e6e8103a031ea))? What's the connection with the stated purpose of the PR?

Will `event` ever be null?

Comment (johnnovak) [reply src/gui/sdl_mapper.cpp:269]: Well, I just reverted all commits from that PR. Check out the commit message of this commit in the linked PR for an explanation.

**Review by weirddan455 — APPROVED** [review]
I agree with reverting this to fix the regressions.

File: src/gui/sdl_mapper.cpp (hunk 1 above)
Comment (arifogel) [reply src/gui/sdl_mapper.cpp:269]: That's fair, but I was just curious why that `event` null check early exit was needed and whether removing it re-adds some bug. That being said, such an analysis is certainly not a blocker for a revert of a known regression.

**Comment by johnnovak** [comment]
> Should "re-opens #4051" be added as linked Issue?

Yeah, good call.

File: src/gui/sdl_mapper.cpp (hunk 1 above)
Comment (johnnovak) [reply src/gui/sdl_mapper.cpp:269]: Seems like it was needed for the equality checks. But it's the mapper, so anything goes 😅

**Pull request merged**